src/Spice86.Core/Emulator/Devices/Video/Renderer.cs

```diff
@@ -289,30 +295,42 @@ public class Renderer : IVgaRenderer {
         uint pixel34Color = GetDacPaletteColor(plane1);
         uint pixel56Color = GetDacPaletteColor(plane2);
         uint pixel78Color = GetDacPaletteColor(plane3);
-        frameBuffer[destinationAddress++] = pixel12Color;
-        frameBuffer[destinationAddress++] = pixel12Color;
-        frameBuffer[destinationAddress++] = pixel34Color;
-        frameBuffer[destinationAddress++] = pixel34Color;
-        frameBuffer[destinationAddress++] = pixel56Color;
-        frameBuffer[destinationAddress++] = pixel56Color;
-        frameBuffer[destinationAddress++] = pixel78Color;
-        frameBuffer[destinationAddress++] = pixel78Color;
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel12Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel12Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel34Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel34Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel56Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel56Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel78Color);
+        TryWriteElementAt(frameBuffer, destinationAddress++, pixel78Color);
+    }
+    
+    private static int? GetElementAtOrNull(IList<byte> collection, int index) {
+        if (index < collection.Count) {
+            return collection[index];
+        }
+        return null;
     }
 
     private uint GetDacPaletteColor(int index) {
         switch (_state.AttributeControllerRegisters.AttributeControllerModeRegister.PixelWidth8) {
             case true:
                 return _state.DacRegisters.ArgbPalette[index];
             default: {
-                int fromPaletteRam6Bits = _state.AttributeControllerRegisters.InternalPalette[index];
-                int bits0To3 = fromPaletteRam6Bits & 0b00001111;
-                int bits4And5 = _state.AttributeControllerRegisters.AttributeControllerModeRegister.VideoOutput45Select
-                    ? _state.AttributeControllerRegisters.ColorSelectRegister.Bits45 << 4
-                    : fromPaletteRam6Bits & 0b00110000;
-                int bits6And7 = _state.AttributeControllerRegisters.ColorSelectRegister.Bits67 << 6;
-                int dacIndex8Bits = bits6And7 | bits4And5 | bits0To3;
-                int paletteIndex = dacIndex8Bits & _state.DacRegisters.PixelMask;
-                return _state.DacRegisters.ArgbPalette[paletteIndex];
+                int? fromPaletteRam6BitsOrNull = GetElementAtOrNull(_state.AttributeControllerRegisters.InternalPalette, index);
```

Comment:
This appears to be a fix for a whole different problem. One that has nothing to do with resoluation changes.
If the `AttributeControllerModeRegister.PixelWidth8` bit is False, then "index" should be a 4-bit number.

I have so far been unable to reproduce this situation so I don't know where the higher index number comes from. But I think finding and fixing that would be better than to hide the bug this way.

Perhaps this is also a timing issue, where the register has already been disabled while it is still reading 8-bit data.
In that case I should see if we maybe need to latch the register at the start of the frame or the start of the row.

Reply:
I'm way out of my depth, I'm afraid. What do you mean by latching ?